flake.nix

```diff
@@ -0,0 +1,41 @@
+# SPDX-FileCopyrightText: 2021 Serokell <https://serokell.io/>
+#
+# SPDX-License-Identifier: CC0-1.0
+
+{
+  description = "My haskell application";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, flake-utils }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = nixpkgs.legacyPackages.${system};
+
+        haskellPackages = pkgs.haskellPackages;
+
+        jailbreakUnbreak = pkg:
+          pkgs.haskell.lib.doJailbreak (pkg.overrideAttrs (_: { meta = { }; }));
+
+        packageName = "customer-client";
+      in {
+        packages.${packageName} =
+          haskellPackages.callCabal2nix packageName self rec {
```

Comment:
> Is there a specific reason cabal2nix was used over haskell.nix?

I had already working setup with cabal2nix and it better matches nixos caches, so it's easier to use as a dependency.

Reply:
Btw, would be nice to check that this setup works in CI to avoid breaking it accidentally in future.
Also, AFAIR, one downside of using `cabal2nix` is that it will use dependencies from `haskellPackages` from `nixpkgs`, which may be outdated and/or broken, while `haskell.nix` will use deps from hackage directly